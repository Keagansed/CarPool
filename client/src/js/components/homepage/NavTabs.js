import { observer } from 'mobx-react';
import { Link } from 'react-router-dom';
import React, { Component } from 'react';

@observer class NavTabs extends Component{

    constructor(){
        super()

        this.state = {
            routeTabActive: "active",
            carpoolTabActive: "",
            tripTabActive: "",    
        }
    }

    handleRouteToggle = (event) => {
        event.preventDefault();

        this.props.store.toggleToRoute();
        this.setState({
            routeTabActive: "active",
            carpoolTabActive: "",
            tripTabActive: "", 
        })
    }

    handleCarPoolToggle = (event) => {
        event.preventDefault();

        this.props.store.toggleToCarpool();
        this.setState({
            routeTabActive: "",
            carpoolTabActive: "active",
            tripTabActive: "", 
        })
    }

    handleTripToggle = (event) => {
        event.preventDefault();

        this.props.store.toggleToTrip();
        this.setState({
            routeTabActive: "",
            carpoolTabActive: "",
            tripTabActive: "active", 
        })
    }

    render(){
        return(
        
            <ul className="nav nav-tabs bg-primary fixed">
                <li className="nav-item w-25 bg-info text-center px-0">
                    <a href="" className={"nav-link bg-info " + this.state.routeTabActive} onClick={this.handleRouteToggle}>
                        <b>
                            <b>Routes</b>
                        </b>
                    </a>
                </li>
                <li className="nav-item w-25 text-center bg-info">
                    <a className={"nav-link bg-info " + this.state.carpoolTabActive} href="" onClick={this.handleCarPoolToggle}>
                        <b>
                            <b>Carpools</b>
                        </b>
                    </a>
                </li>
                <li className="nav-item w-25 text-center bg-info">
                    <a className={"nav-link bg-info " + this.state.tripTabActive} href="" onClick={this.handleTripToggle}>
                        <b>
                            <b>Trips</b>
                        </b>
                    </a>
                </li>
                <li className="nav-item w-25 bg-info text-center flex-center no-active">
                    <div className="nav-link p-0 bg-info no-active" href="">
                        <Link className="text-secondary" to={{pathname: '/newRoute', token: this.props.token}}>
                            <i className="fa fa-plus fa-lg fa-fw d-inline nav-item-circle" onClick={this.handleAddRoute}></i>
                        </Link>
                    </div>
                </li>
            </ul>
           
        )
    }
}

export default NavTabs