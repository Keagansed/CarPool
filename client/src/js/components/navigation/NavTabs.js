// import { Link } from 'react-router-dom';
import { observer } from 'mobx-react';
import React, { Component } from 'react';
import RoutesStore from "../../stores/RoutesStore";

import NewRouteModal from '../route/NewRouteModal';

@observer class NavTabs extends Component{

    constructor(){
        super()

        this.state = {
            routeTabActive: "active",
            carpoolTabActive: "",
            tripTabActive: "",    
        }
    }

    handleRouteToggle = (event) => {
        event.preventDefault();

        this.props.store.toggleToRoute();
        this.setState({
            routeTabActive: "active",
            carpoolTabActive: "",
            tripTabActive: "", 
        })
    }

    handleCarPoolToggle = (event) => {
        event.preventDefault();

        this.props.store.toggleToCarpool();
        this.setState({
            routeTabActive: "",
            carpoolTabActive: "active",
            tripTabActive: "", 
        })
    }

    handleTripToggle = (event) => {
        event.preventDefault();

        this.props.store.toggleToTrip();
        this.setState({
            routeTabActive: "",
            carpoolTabActive: "",
            tripTabActive: "active", 
        })
    }

    render(){
        return(
            <div className="fixed-top container-fluid height-50px bg-aqua">
                <div className="row font-20px height-100p">
                    <div className="col-10 pad-0">
                        <button className={"btnTab col-4 btn height-100p bg-trans txt-purple fw-bold brad-0 font-20px " + this.state.routeTabActive} onClick={this.handleRouteToggle}>
                            Routes
                        </button>
                        <button className={"btnTab col-4 btn height-100p bg-trans txt-purple fw-bold brad-0 font-20px " + this.state.carpoolTabActive} onClick={this.handleCarPoolToggle}>
                            Carpools
                        </button>
                        <button className={"btnTab col-4 btn height-100p bg-trans txt-purple fw-bold brad-0 font-20px " + this.state.tripTabActive} onClick={this.handleTripToggle}>
                            Trips
                        </button>
                    </div>
                    <NewRouteModal store={RoutesStore} token={this.props.store.token}/>
                </div>
            </div>
        )
    }
}

export default NavTabs;