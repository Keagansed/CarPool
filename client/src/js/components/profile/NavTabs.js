// File Type: Component

import React, { Component } from 'react';
import { observer } from 'mobx-react';

/*
* The purpose of this NavTabs class is to provide a tabular menu that is available on the Profile page.
* This tabbed menu provides access to the "Vouches" and "Trust" tabs on the Profile Page.
*/
@observer class NavTabs extends Component{

    /*
    * The purpose of the constructor method is to instantiate fields to relevant values. In this case
    * the vouchTabActive field is set to true because the "Vouches" tab is active when the Profile page is first
    * loaded.
    */
    constructor(){
        super()

        this.state = {
            //vouchTabActive represents the status of the "Vouches" tab.
            vouchTabActive: "active",
            //trustTabActive represents the status of the "Trust" tab.
            trustTabActive: ""   
        }
    }

    /*
    * The purpose of the handleVouchToggle method is to set the "Vouches" tab to active. The method
    * will also deactivate the other tabs.
    */
    handleVouchToggle = (event) => {
        event.preventDefault();

        this.props.store.toggleToVouch();
        this.setState({
            vouchTabActive: "active",
            trustTabActive: ""
        });
    }

    /*
    * The purpose of the handleTrustToggle method is to set the "Trust" tab to active. The method
    * will also deactivate the other tabs.
    */
    handleTrustToggle = (event) => {
        event.preventDefault();

        this.props.store.toggleToTrust();
        this.setState({
            vouchTabActive: "",
            trustTabActive: "active"
        });
    }

    /*
    * The purpose of the render method is to enable the rendering of this component.
    * It returns react elements and HTML using JSX.
    */
    render(){
        return(
            <div className="container-fluid height-50px bg-aqua">
                <div className="row font-20px height-100p">
                    <button 
                        className={"btnTabWhiteActive col-6 btn height-100p bg-trans txt-purple fw-bold brad-0 font-20px " + this.state.vouchTabActive} 
                        onClick={this.handleVouchToggle}
                    >
                        Vouches
                    </button>
                    <button 
                        className={"btnTabWhiteActive col-6 btn height-100p bg-trans txt-purple fw-bold brad-0 font-20px " + this.state.trustTabActive} 
                        onClick={this.handleTrustToggle}
                    >
                        Trust
                    </button>
                </div>
            </div>
        )
    }
}

export default NavTabs;