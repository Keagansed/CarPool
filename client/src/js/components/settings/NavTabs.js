import { observer } from 'mobx-react';
import React, { Component } from 'react';

@observer class NavTabs extends Component{

    constructor(){
        super()

        this.state = {
            profileTab: "active",
            alertsTab: "",    
        }
    }

    handleProfileToggle = (event) => {
        event.preventDefault();

        this.props.store.toggleToProfile();
        this.setState({
            profileTab: "active",
            alertsTab: "", 
        })
    }

    handleAlertsToggle = (event) => {
        event.preventDefault();

        this.props.store.toggleToAlerts();
        this.setState({
            profileTab: "",
            alertsTab: "active", 
        })
    }

    render(){
        return(
            <div className="fixed-top container-fluid height-50px bg-aqua">
                <div className="row font-20px height-100p">
                    <button className={"btnTab  col-6 btn height-100p bg-trans txt-purple fw-bold brad-0 font-20px " + this.state.profileTab} onClick={this.handleProfileToggle}>
                        Profile
                    </button>
                    <button className={"btnTab  col-6 btn height-100p bg-trans txt-purple fw-bold brad-0 font-20px " + this.state.alertsTab} onClick={this.handleAlertsToggle}>
                        Alerts
                    </button>
                </div>
            </div>
        )
    }
}

export default NavTabs;