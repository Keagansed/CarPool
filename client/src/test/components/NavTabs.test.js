import React from 'react';
import NavTabs from '../../js/components/home/NavTabs';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';


describe('NavTabs Component', () => {
    let container;

    beforeEach(() => {
        container = shallow(<NavTabs />);
    });

    it('captures snapshot', () => {
        const wrapper = renderer.create(<NavTabs />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly', () => {
        expect(container.length).toEqual(1);
    });
});
