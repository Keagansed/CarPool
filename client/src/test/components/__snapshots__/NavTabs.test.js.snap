// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavTabs Component captures snapshot 1`] = `
<div
  className="fixed-top container-fluid height-50px bg-aqua"
>
  <div
    className="row font-20px height-100p"
  >
    <div
      className="col-10 pad-0"
    >
      <button
        className="btnTab col-4 btn height-100p bg-trans txt-purple fw-bold brad-0 font-20px active"
        onClick={[Function]}
      >
        Routes
      </button>
      <button
        className="btnTab col-4 btn height-100p bg-trans txt-purple fw-bold brad-0 font-20px "
        onClick={[Function]}
      >
        Carpools
      </button>
      <button
        className="btnTab col-4 btn height-100p bg-trans txt-purple fw-bold brad-0 font-20px "
        onClick={[Function]}
      >
        Trips
      </button>
    </div>
    <div
      className="mx-auto"
    >
      <button
        className="col-2 btn height-100p bg-trans txt-purple fw-bold brad-0 font-20px txt-center"
        onClick={[Function]}
      >
        <i
          className="fa fa-plus"
        />
      </button>
      <div
        className="modal"
        id="myModal"
        role="dialog"
        style={
          Object {
            "display": "none",
          }
        }
        tabIndex="-1"
      >
        <div
          className="modal-dialog"
          role="document"
        >
          <div
            className="modal-content"
          >
            <div
              className="modal-header bg-aqua"
            >
              <h5
                className="modal-title fw-bold"
              >
                Create a New Route
              </h5>
              <button
                aria-label="Close"
                className="close"
                onClick={[Function]}
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  Ã—
                </span>
              </button>
            </div>
            <div
              className="modal-body"
            >
              <form>
                <div
                  className="row"
                >
                  <h6
                    className="fw-bold mx-auto"
                  >
                    Time and Date
                  </h6>
                </div>
                <div
                  className="row padbot-10px"
                >
                  <input
                    className="col-5 form-control mx-auto brad-2rem"
                    id="inputRouteTime"
                    name="Time"
                    placeholder="Time"
                    required="required"
                    type="time"
                  />
                  <input
                    className="col-5 form-control mx-auto brad-2rem"
                    id="inputRouteDate"
                    name="Date"
                    placeholder="Date"
                    required="required"
                    type="date"
                  />
                </div>
                <div
                  className="row"
                >
                  <h6
                    className="fw-bold mx-auto"
                  >
                    Repeat Weekly
                  </h6>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="mx-auto"
                  >
                    <div
                      className="weekDays-selector"
                    >
                      <input
                        className="weekday"
                        id="weekday-mon"
                        type="checkbox"
                      />
                      <label
                        htmlFor="weekday-mon"
                      >
                        M
                      </label>
                      <input
                        className="weekday"
                        id="weekday-tue"
                        type="checkbox"
                      />
                      <label
                        htmlFor="weekday-tue"
                      >
                        T
                      </label>
                      <input
                        className="weekday"
                        id="weekday-wed"
                        type="checkbox"
                      />
                      <label
                        htmlFor="weekday-wed"
                      >
                        W
                      </label>
                      <input
                        className="weekday"
                        id="weekday-thu"
                        type="checkbox"
                      />
                      <label
                        htmlFor="weekday-thu"
                      >
                        T
                      </label>
                      <input
                        className="weekday"
                        id="weekday-fri"
                        type="checkbox"
                      />
                      <label
                        htmlFor="weekday-fri"
                      >
                        F
                      </label>
                      <input
                        className="weekday"
                        id="weekday-sat"
                        type="checkbox"
                      />
                      <label
                        htmlFor="weekday-sat"
                      >
                        S
                      </label>
                      <input
                        className="weekday"
                        id="weekday-sun"
                        type="checkbox"
                      />
                      <label
                        htmlFor="weekday-sun"
                      >
                        S
                      </label>
                    </div>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <h6
                    className="fw-bold mx-auto"
                  >
                    Start and End Locations
                  </h6>
                </div>
                <div
                  className="row padbot-10px"
                >
                  <input
                    className="col-11 form-control mx-auto brad-2rem"
                    id="inputRouteStart"
                    name="StartLocation"
                    placeholder="Start Location"
                    required="required"
                    type="text"
                  />
                </div>
                <div
                  className="row padbot-10px"
                >
                  <input
                    className="col-11 form-control mx-auto brad-2rem"
                    id="inputRouteEnd"
                    name="EndLocation"
                    placeholder="End Location"
                    required="required"
                    type="text"
                  />
                </div>
                <div
                  className="row"
                >
                  <button
                    className="btn btn-primary mx-auto width-15rem brad-2rem mbottom-0 bg-aqua txt-purple fw-bold"
                    id="btnNewRoute"
                    type="submit"
                  >
                    Add Route
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
